tensorflow/python/ops/array_ops.py

```diff
@@ -1273,6 +1277,57 @@ def transpose(a, perm=None, name="transpose"):
     return ret
 
 
+def framesig(sig, frame_len, frame_step, parallel=False, name="framesig"):
+  """Frame a signal into overlapping frames.
+  May be used in front of spectral functions.
+  
+  For example:
+  
+  ```python
+  pcm = tf.placeholder(tf.float32, [None, 9152])
+  frames = tf.framesig(pcm, 512, 180)
+  magspec = tf.abs(tf.spectral.rfft(frames, tf.constant(512, shape=[1])))
+  image = tf.reshape(magspec, [-1, 49, 257, 1])
+  ```
+  
+  Args:
+    sig: A `Tensor` with `rank = 2`.
+    frame_len: The length of each frame.
+    frame_step: The step between frames.
+    parallel: Whether to use `tf.parallel_stack` or `tf.stack`
+    name: A name for the operation (optional).
+  
+  Returns:
+    A `Tensor` of frames with shape [batch_size, num_frames, frame_len].
+  """
+  sig_shape = sig.get_shape()
```

Comment:
this may be TensorShape(None) - in which case, neither the rank nor the batch size, nor the length are known.  in that case you have to do all your calculations using tensorflow math_ops.

you could require that sig_shape.ndims == 2, which is fine; but len(sig_shape) will not work.